File: Java/프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    public int solution(int[] arr) {
        int answer = arr[0];
        for (int number : arr) {
            answer = (answer * number) / euclid(number, answer);
        }
        
        return answer;
    }
    
    private int euclid(int a, int b) {
        while (b != 0) {
            int r = a % b;
            a = b;
            b = r;
        }
        return a;
    }
}

File: Swift/프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.swift
func solution(_ arr:[Int]) -> Int {
    var answer = arr[0]
    for num in arr {
        answer = (num * answer) / gcd(num, answer)
    }
    return answer
}

fileprivate func gcd(_ a: Int, _ b: Int) -> Int {
    var aCopy = a
    var bCopy = b
    while bCopy != 0 {
        let r = aCopy % bCopy
        aCopy = bCopy
        bCopy = r
    }
    
    return aCopy
}